diesel/src/expression/mod.rs

```diff
@@ -1,14 +1,14 @@
 //! AST types representing various typed SQL expressions. Almost all types
-//! implement either [`Expression`](trait.Expression.html) or
-//! [`AsExpression`](trait.AsExpression.html).
+//! implement either [`Expression`](/diesel/expression/trait.Expression.html) or
```

Comment:
The relative links don't work when the links appears on multiple pages with different roots. For example with the current setup `Expression` and `AsExpression` work [here](https://docs.rs/diesel/0.9.0/diesel/expression/index.html) but break [here](https://docs.rs/diesel/0.9.0/diesel/). The opposite will happen if the links were `expression/trait.Expression.html` and `expression/trait.AsExpression.html`.

So you are right that there is a need to refer to the root of the package, without hard coding it.